chess/src/castling_rights.rs

```diff
@@ -139,4 +139,25 @@ impl CastlingRights {
     pub fn has_castling_right(&self, right: CastlingRightsKind) -> bool {
         self.0[right as usize]
     }
+
+    pub fn get_fen_str(&self) -> String {
```

Comment:
could we make use of `with_capacity` here also somehow? Also not sure if there's a way to do it less pushes using `format!` (though `format!` probably does some pushing underneath so it probably doesn't matter)